source/source-connector/usage-examples/topic-naming.txt

```diff
@@ -113,14 +123,121 @@ mapping, the connector performs the following actions:
+--------------------------------------------
+
+You can use a regular expression (regex) within a
+topic namespace map. To use a regular expression, start the namespace
+pattern with the forward slash (``/``) character. Create the regular
+expression following the syntax and behavior specified by the
+``java.util.regex.Pattern`` class.
+
+.. important::
+   
+   Because the ``/`` character denotes that the namespace is a
+   regex pattern, the connector raises a ``ConnectConfigException``
+   if the namespace includes this character in a non-regex context.
+
+The connector computes the topic name by doing variable expansion on the
+topic name template. The connector supports the following variables: 
+ 
+- ``db``: The database name from the matching namespace.
+- ``sep``: The value of the ``topic.separator`` configuration property.
+  To learn more about this property, see :ref:`source-configuration-kafka-topic`.
+- ``coll``: The collection name from the matching namespace, or an empty
+  string if there is no collection name.
+- ``sep_coll``: The value of the ``coll`` variable prefixed with the value
+  of the ``sep`` variable, or an empty string if the value of ``coll`` is empty.
+- ``coll_sep``: The value of the ``coll`` variable suffixed with the
+  value of the ``sep`` variable, or an empty string if the value of ``coll`` is empty.
+- ``sep_coll_sep``: The value of the ``coll`` variable prefixed and
+  suffixed with the value of the ``sep`` variable, or an empty string if
+  the value of ``coll`` is empty.
+ 
+If you use any of the previous variables in your topic name template,
+you must enclose them in curly brackets (``{}``) for the connector to
+expand them. You can't use curly brackets in the topic name template for
+any other purpose.
+
+.. note:: Escaping Characters
+
+   If your namespace pattern includes characters that need escaping,
+   you must prefix these characters with two backslashes (``\\``) in the
+   pattern. For example, to match a period (``'.'``) character, you must
+   write it as ``'\\.'``.
+ 
+The connector matches namespaces in the following order:
+ 
+1. Pairs with database and collection names in the namespace pattern. To
+   learn more about this namespace pattern, see the :ref:`Topic
+   Namespace Map Example <topic-naming-namespace-map-example>`.
+#. Pairs with only a database name in the namespace pattern. To
+   learn more about this namespace pattern, see the :ref:`Topic
+   Namespace Map Example <topic-naming-namespace-map-example>`.
+#. Regex pairs in order.
+#. The wildcard pair. To learn more about this namespace pattern, see
+   the :ref:`Topic Namespace Map with Wildcard Example
+   <topic-naming-namespace-map-wildcard-example>`.
+
+This example shows how to specify the ``topic.namespace.map``
+setting so that the connector performs the following mapping:
+
+- Writes events from databases that match a regular expression to the
+  ``industrial{sep_coll}`` topic. The regex pattern matches any database
```

Comment:
I did not mention the value of `topic.separator` because `.` is the default, can fix.
I don't understand the contradiction you pointed out.

I thought if the collection name was `something`, then {sep_coll} would be expanded to `.something` (default separator). Is this not what I described?

In any case, I am going to use the example from the ticket instead, as you requested in your next comment.

Reply:
> I thought if the collection name was something, then {sep_coll} would be expanded to .something (default separator).

Exactly, which means that `industrial{sep_coll}` is not a topic name, as it is stated in the docs at the moment. But because the distinction between a topic name template and a topic name is not done in the docs, we had the following two lines in the PR when I this thread was started:

> ... to the ``industrial{sep_coll}`` topic.
> ... to the ``industrial.<collectionName>`` topic.

The first one is a topic name template, the second one is, indeed, a topic name<sup>1</sup>. If the documentation makes the distinction, one will be able to see why the text in those documentation lines is different. I tried to expand more on this in https://github.com/mongodb/docs-kafka-connector/pull/141#discussion_r1349922059, where a similar issue exists in a couple of different places. If I failed to get a point across again, I'll not push this issue further in either of these threads.

---

<sup>1</sup> Strictly speaking, one may argue that `industrial.<collectionName>` is also template, and in a way it is: it shows what is computed based on the topic name template `industrial{sep_coll}` without focusing on a specific collection name. But even if so, `industrial.<collectionName>` is not the same thing as a template from pairs in `topic.namespace.map`.